src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -32,193 +32,614 @@
-                badTargetFieldAPIName || 
-                badTargetObjectMappingId || 
-                badIsDeletedField;
+    /*******************************************************************************************************
+    * @description Wrapper class for Data_Import_Field_Mapping_Set__mdt used to create
+    * the custom CMT data as well as for use in the Custom UI.
+    */
+    public class DataImportFieldMappingSet {
+        public String recordName;
+        public String label;
+        public String dataImportObjectMappingSetRecordName;
+
+        public DataImportFieldMappingSet(String label, String dataImportObjectMappingSetRecordName) {
+            this.label = label.trim();
+            this.dataImportObjectMappingSetRecordName = dataImportObjectMappingSetRecordName;
+        }
+
+        public Metadata.CustomMetadata getMetadataRecord() {
+            String mdTypeName = 'Data_Import_Field_Mapping_Set__mdt';
 
-        return foundMismatch;
+            // Setup custom metadata to be created or updated in the subscriber org.
+            Metadata.CustomMetadata customMd = new Metadata.CustomMetadata();
+            customMd.label = this.label.left(40);
+            this.recordName = this.label;
+            customMd.fullName = mdTypeName + '.' + this.recordName.left(40).removeEnd('_');
+            customMd.protected_x = false;
+
+            Map<String, Object> fieldValues = new Map<String, Object>{
+                'Data_Import_Object_Mapping_Set__c' => this.dataImportObjectMappingSetRecordName
+            };
+            customMd = CMT_MetadataAPI.addFieldsToMetadataObject(customMd, fieldValues);
+
+            return customMd;
+        }
     }
 
-    /** @description Name of default packaged Object Mapping Set */
-    private static final String DEFAULT_OBJECT_MAPPING_SET_RECORD_NAME = 'Default_Object_Mapping_Set';
+    /*******************************************************************************************************
+    * @description Wrapper class for Data_Import_Field_Mapping__mdt used to create
+    * the default CMT data as well as for use in the Custom UI.
+    */
+    public class DataImportFieldMapping {
+        public String recordName;
+        public String label {get; set;}
+        public String dataImportObjectName {get; set;}
+        public String dataImportFieldMappingSetRecordName;
+        public String dataImportFieldMappingSetLabel;
+        public String sourceFieldAPIName;
+        public String targetFieldAPIName {get; set;}
+        public String targetObjectMappingRecordName;
+        public String targetObjectMappingLabel;
+        public String targetObjectAPIName;
+        public String isRequired;
+        public Boolean isDeleted;
+        public String importedRecordField;
+        public String importedRecordStatusField;
+
+        public DataImportFieldMapping(String label, String sourceFieldAPIName, String dataImportObjectName,
```

Comment:
This is the signature that can be updated to accept two objects as opposed to these 5 values.